main/com.spotify/manifest.tt

```diff
@@ -271,6 +271,70 @@ class @com.spotify
   }]
   #[confirm=enum(auto)];
 
+  action add_song_to_playlist(in req song : Entity(com.spotify:song)
+                                #[string_values="tt:song_name"]
+                                #_[prompt=["what song do you want to add"]]
+                                #_[canonical={
+                                  default="base",
+                                  base=["name"],
+                                  preposition=["named #", "called #", "titled #"],
+                                  property=["name #", "title #"]
+                                }],
+                              in req playlist: String
+                                #[string_values="com.spotify:playlist"]
+                                #_[prompt=["what playlist do you want to add to"]]
+                                #_[canonical={
+                                  default="base",
+                                  base=["name"],
+                                  preposition=["named #", "called #", "titled #"],
+                                  property=["name #", "title #"]
+                                }])
+  #_[canonical=["add a song to a playlist", "add a track to a playlist"]]
+  #_[confirmation="add the song $song to the playlist $playlist"]
+  #_[on_error={
+    disallowed_action=["you can't add songs to this playlist"],
+    rate_limit_error=["spotify is temporarily unavailable"]
+  }]
+  #[confirm=enum auto]
+  #[minimal_projection=[]];
+
+  action add_this_song_to_playlist(in req playlist: String
+                                    #[string_values="com.spotify:playlist"]
+                                    #_[prompt=["what playlist do you want to add to"]]
+                                    #_[canonical={
+                                      default="base",
+                                      base=["name"],
+                                      preposition=["named #", "called #", "titled #"],
+                                      property=["name #", "title #"]
+                                    }])
```

Comment:
Do we need a separate function?
"add the currently playing song to the playlist X" should be:
```
@com.spotify.get_currently_playing() => @com.spotify.add_song_to_playlist(song=song, playlist="X");
```